src/DependencyInjection/Configuration.php

```diff
@@ -140,9 +140,25 @@ public function getConfigTreeBuilder()
                             ->defaultValue($this->debug)
                         ->end()
                         ->scalarNode('formatter')->defaultNull()->end()
-                        ->integerNode('captured_body_length')
+                        ->scalarNode('captured_body_length')
+                            ->beforeNormalization()
+                                ->always(function ($maxLength) {
+                                    if (null === $maxLength) {
+                                        return null;
+                                    }
+
+                                    if (!is_int($maxLength)) {
+                                        $invalidConfiguration = new InvalidConfigurationException('The child node "captured_body_length" at path "httplug.profiling" must be an integer or null.');
```

Comment:
Sorry for being late, but wouldn't it be more straight-forward to use something like `ifTrue(...)->thenInvalid()` construct?

What I have in mind would be something like this:

```php
->scalarNode('captured_body_length')
    ->validate()
        ->ifTrue(function ($v) {
            return null !== $v && !is_int($v);
        })
        ->thenInvalid('The child node "captured_body_length" at path "httplug.profiling" must be an integer or null ("%s" given).')
    ->end()
->end()
```

Reply:
agreed. i created #323 with your proposed change